iree/hal/vmla/vmla_module.h

```diff
@@ -12,25 +12,121 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// NOTE: unlike most VM modules we are only ever created and used from C++ code
+// linked into the same library, because of this we can avoid the C shims and
+// directly use C++ types.
```

Comment:
Correct: it's an implementation detail of the backend and not meant to be exposed to any other code (one of the reasons why it is under iree/hal/vmla/ instead of iree/modules/vmla/ with other shared modules).